tests/performance/test_performance.py

```diff
@@ -54,15 +63,177 @@ def test_performance(config_path):
+
+    If output path is not given, the test case is created with the filename
+    ``{dataset_generator}_{fit_size}_{transform_size}.json`` inside the folder
+    ``{transformer_module}/{transformer_class_name}``.
+
+    Args:
+        transformer_class (type):
+            Class of the transformer to use.
+        tranformer_kwargs (dict):
+            Keyword arguments to pass to the transformer.
+        dataset_generator (type):
+            Dataset Generator class.
+        fit_size (int):
+            Number of values to use when fitting the transformer.
+        transform_size (int):
+            Number of values to use when transforming and reverse transforming.
+        iterations (int):
+            Number of iterations to perform.
+        output_path (str):
+            Optional. Path where the output JSON file is written.
+        config_name (str):
+            Name that should be given to this kwargs config. If not given,
+            and default args are used, name is ``default``. Otherwise, a
+            hash is taken.
+    """
+    transformer_instance = transformer_class(**transformer_kwargs)
+    outputs = find_transformer_boundaries(transformer_instance, dataset_generator,
+                                          transform_size, fit_size, iterations)
+    test_case = {
+        'dataset': get_fqn(dataset_generator),
+        'transformer': get_fqn(transformer_class),
+        'kwargs': transformer_kwargs,
+        'fit_size': fit_size,
+        'transform_size': transform_size,
+        'expected': {
+            'fit': {
+                'time': outputs['Fit Time'],
+                'memory': outputs['Fit Memory'],
+            },
+            'transform': {
+                'time': outputs['Transform Time'],
+                'memory': outputs['Transform Memory'],
+            },
+            'reverse_transform': {
+                'time': outputs['Reverse Transform Time'],
+                'memory': outputs['Reverse Transform Memory'],
+            },
+        }
+    }
+    if output_path is None:
+        if config_name:
+            config_str = config_name
+        elif transformer_kwargs:
+            config_hash = hashlib.md5()
+            config_hash.update(json.dumps(sorted(transformer_kwargs)).encode())
+            config_str = config_hash.hexdigest()[0:8]
+        else:
+            config_str = 'default'
+
+        file_name = f'{config_str}_{dataset_generator.__name__}_{fit_size}_{transform_size}.json'
+        module_name = transformer_class.__module__.rsplit('.', 1)[1]
+        output_path = TEST_CASES_PATH / module_name / transformer_class.__name__ / file_name
+
+    os.makedirs(output_path.parent, exist_ok=True)
+    with open(output_path, 'w') as output_file:
+        json.dump(test_case, output_file, indent=4)
+
+    cwd = os.getcwd()
+    if str(output_path).startswith(cwd):
+        output_path = str(output_path)[len(cwd) + 1:]
+
+    print(f'Test case created: {output_path}')
```

Comment:
When do we want to use print statements vs loggers?

Reply:
In general I tend to use `print` statements for quick command line interfaces or simple user-facing functions like this one which are likely to be executed without any logging setup. For any function that is likely to be integrated withing other code or command line interfaces which are complex enough to offer verbosity or output file controls, I use logging.